automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookPage.scala

```diff
@@ -89,38 +94,39 @@ class NotebookPage(override val url: String)(override implicit val authToken: Au
 
   lazy val cells: Query = cssSelector(".CodeMirror")
 
-  def lastCell: WebElement = {
-    webDriver.findElements(cells.by).asScala.toList.last
+  def lastCell: (Int, WebElement) = {
+    val asList = webDriver.findElements(cells.by).asScala.toList
+    (asList.length, asList.last)
   }
 
-
-
   def cellOutput(cell: WebElement): Option[String] = {
     val outputs = cell.findElements(By.xpath("../../../..//div[contains(@class, 'output_subarea')]"))
     outputs.asScala.headOption.map(_.getText)
   }
-
+  
   def executeCell(code: String, timeout: FiniteDuration = 1 minute): Option[String] = {
     await enabled cells
-    val cell = lastCell
+    val (cellNumber, cell) = lastCell
     val jsEscapedCode = StringEscapeUtils.escapeEcmaScript(code)
     executeScript(s"""arguments[0].CodeMirror.setValue("$jsEscapedCode");""", cell)
     click on runCellButton
     await condition (!cellsAreRunning, timeout.toSeconds)
```

Comment:
I'm wondering if this condition is needed anymore? Shouldn't waiting for `cellIsRendered` be sufficient?

Reply:
Probably not.  Could provide additional information for a stack trace though: if we fail here, we know it's specifically because a cell is still running.  "Cell 3 not found" could conceivably be something different.

Reply:
True, 👍